core/drivers/crypto/se050/core/rsa.c

```diff
@@ -15,6 +15,10 @@
 #include <tee/tee_cryp_utl.h>
 #include <tee_api_defines_extensions.h>
 
+#define TEE_OAEP_ALG(__hash_alg) \
+	((TEE_ALG_RSAES_PKCS1_OAEP_MGF1_SHA1 & ~GENMASK_32(23, 20)) | \
+	SHIFT_U32(__hash_alg, 20))
```

Comment:
Since we're only interested in the lowest four bits we should use a mask for that.

Reply:
um ok, but it is unnecessary code since the hash_algo must have zeros in those values anyway.

Reply:
Yeah, just as unnecessary as GENMASK_32 above where it's only a single bit that you need to clear.

Reply:
nah, that is silly: the one you want added does nothing while the one that I added is required (even is clearing a couple of bits that will be zeros anyway).
but anyway, I am going to remove the macro altogether since only SHA1 seems to be required according to latest spec. - though that might change in the future. 

https://www.nxp.com/docs/en/application-note/AN12413.pdf (page 40)